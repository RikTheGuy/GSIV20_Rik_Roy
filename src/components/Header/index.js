import React from 'react'
import { Link } from 'react-router-dom'
import Searchbar from '../Searchbar/index.js'

import './index.scss'
import 'material-icons/iconfont/filled.css'

const index = ({ title = 'Title', searchbar = false }) => {
    return (
        <header className='header'>
            <nav>
                <div data-testid='title' className='title'>
                    {
                        searchbar ?
                            <Searchbar /> :
                            <h4>{title}</h4>
                    }
                </div>
                <Link to='/'>
                    <span className="material-icons home-icon">home</span>
                </Link>
            </nav>
        </header>
    )
}

export default index
