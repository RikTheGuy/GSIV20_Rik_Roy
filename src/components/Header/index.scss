@import "../../styles/colors.scss";
@import "../../styles/values.scss";

.header {
  background-color: $primary;
  color: $accent;

  nav {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    box-shadow: 0 0.1rem 0.3rem 0.1rem $light-gray;
    padding: 1rem 1rem;

    a {
      color: inherit;
    }

    @media screen and (min-width: $breakpoint) {
      padding: 1rem 3rem;
    }

    .title {
      width: 75%;
      @media screen and (min-width: $breakpoint) {
        width: 400px;
      }
      h4 {
        margin: 0;
        font-weight: 600;
        font-size: 1rem;
      }
    }
  }
}
