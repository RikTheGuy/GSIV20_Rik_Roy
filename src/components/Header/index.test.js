import { BrowserRouter as Router } from 'react-router-dom';
import Header from './index.js';
import { render, cleanup } from '@testing-library/react';

import renderer from 'react-test-renderer';

describe('Header', () => {
    afterEach(cleanup)
    describe('Renders Successfully', () => {
        test('Title Renders', () => {
            const { getByTestId } = render(
                <Router>
                    <Header title='Sample Title' />
                </Router>)
            expect(getByTestId('title')).toHaveTextContent('Sample Title')
        })

        test('Contains Searchbar', () => {
            const { getByTestId } = render(
                <Router>
                    <Header title='Sample Title' searchbar />
                </Router>)
            expect(getByTestId('title')).toContainElement(getByTestId('searchbar'))
        })

        test('Title Does not show on Searchbar', () => {
            const { getByTestId } = render(
                <Router>
                    <Header title='Sample Title' searchbar />
                </Router>)
            expect(getByTestId('title')).not.toHaveTextContent('Sample Title')
        })
    })

    describe('Snapshot Testing', () => {
        test('Test With Title', () => {
            const tree = renderer.create(<Router><Header title='Sample Title' /></Router>).toJSON()
            expect(tree).toMatchSnapshot()
        })

        test('Test With Searchbar', () => {
            const tree = renderer.create(<Router><Header searchbar /></Router>).toJSON()
            expect(tree).toMatchSnapshot()
        })

        test('Test With Searchbar and title', () => {
            const tree = renderer.create(<Router><Header searchbar title='Something' /></Router>).toJSON()
            expect(tree).toMatchSnapshot()
        })
    })
})