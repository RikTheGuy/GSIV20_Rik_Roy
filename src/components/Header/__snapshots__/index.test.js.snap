// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Snapshot Testing Test With Searchbar 1`] = `
<header
  className="header"
>
  <nav>
    <div
      className="title"
      data-testid="title"
    >
      <div
        className="searchbar"
        data-testid="searchbar"
      >
        <span
          className="material-icons icon"
        >
          search
        </span>
        <input
          id="search-input"
          placeholder="Search"
          type="text"
        />
      </div>
    </div>
    <a
      href="/"
      onClick={[Function]}
    >
      <span
        className="material-icons home-icon"
      >
        home
      </span>
    </a>
  </nav>
</header>
`;

exports[`Header Snapshot Testing Test With Searchbar and title 1`] = `
<header
  className="header"
>
  <nav>
    <div
      className="title"
      data-testid="title"
    >
      <div
        className="searchbar"
        data-testid="searchbar"
      >
        <span
          className="material-icons icon"
        >
          search
        </span>
        <input
          id="search-input"
          placeholder="Search"
          type="text"
        />
      </div>
    </div>
    <a
      href="/"
      onClick={[Function]}
    >
      <span
        className="material-icons home-icon"
      >
        home
      </span>
    </a>
  </nav>
</header>
`;

exports[`Header Snapshot Testing Test With Title 1`] = `
<header
  className="header"
>
  <nav>
    <div
      className="title"
      data-testid="title"
    >
      <h4>
        Sample Title
      </h4>
    </div>
    <a
      href="/"
      onClick={[Function]}
    >
      <span
        className="material-icons home-icon"
      >
        home
      </span>
    </a>
  </nav>
</header>
`;
